### PR TITLE
adding visual-diff support based on the GitHub Action version

### PR DESCRIPTION
**Pull request by dlockhart**
Switches this to use the new GHA-based visual diff.

### CODE REVIEW

File: src/generators/demo/templates/index.html [hunk 1]
```diff
@@ -9,7 +9,6 @@
 		</script>
 		<title><%= className %></title>
 		<meta name="viewport" content="width=device-width, initial-scale=1.0">
-		<meta http-equiv="X-UA-Compatible" content="ie=edge">
```
Comment (dlockhart) [inline src/generators/demo/templates/index.html:12]: These aren't needed anymore.

File: src/generators/release/index.js [hunk 2]
```diff
@@ -1,10 +1,13 @@
-import { getDestinationPath, mergeText, moveFilesInDir } from '../../helper.js';
+import { getDestinationPath, mergeText, moveFile } from '../../helper.js';
 
 export function run(templateData) {
 	mergeText(
 		`${__dirname}/templates/configured/_README.md`,
 		`${getDestinationPath(templateData.hyphenatedName)}/README.md`
 	);
 
-	moveFilesInDir(`${__dirname}/templates/static`, getDestinationPath(templateData.hyphenatedName));
```
Comment (dlockhart) [inline src/generators/release/index.js:9]: Multiple stages couldn't statically move the directory `.github/workflows` so I switched this to just move the individual files.

File: src/generators/test-unit/templates/configured/_package.json [hunk 3]
```diff
@@ -11,6 +11,6 @@
     "@open-wc/testing": "^2",
     "@open-wc/testing-karma": "^4",
     "deepmerge": "^4",
-    "karma-sauce-launcher": "^2"
+    "karma-sauce-launcher": "^4"
```
Comment (dlockhart) [inline src/generators/test-unit/templates/configured/_package.json:14]: AFAIK the issues we were having with version 4 have been resolved.

File: src/generators/test-visual-diff/templates/_element.visual-diff.js [hunk 4]
```diff
@@ -10,7 +10,6 @@ describe('<%= tagName %>', () => {
 	before(async() => {
 		browser = await puppeteer.launch();
 		page = await visualDiff.createPage(browser);
-		await page.setViewport({ width: 800, height: 800, deviceScaleFactor: 2 });
```
Comment (dlockhart) [inline src/generators/test-visual-diff/templates/_element.visual-diff.js:13]: Viewport is automatically configured by `createPage()`

File: README.md [hunk 5]
```diff
@@ -33,19 +33,13 @@ npm init @brightspace-ui
...
 ### Visual Diff Testing
 
-**Note that this will soon be in a Github action**
-
-To do visual diff testing, the `GITHUB_RELEASE_TOKEN`, `VISUAL_DIFF_S3_ID`, and `VISUAL_DIFF_S3_SECRET` need to be encrypted into the .travis.yml file.
-
-To learn more about how to set this up, see the [Running in CI](https://github.com/BrightspaceUI/visual-diff#running-in-ci) section of the visual-diff readme.
-
-To use the Visual Difference Bot to run visual diff testing, see [here](https://github.com/BrightspaceUI/visual-difference-bot/blob/master/README.md/#utilizing-the-deployed-bot-for-your-existing-brightspace-repo).
+Visual diff results are published to a bucket in S3 and need special tokens to do so. To set these up, follow the instructions in the [visuald-iff GitHub Action](https://github.com/BrightspaceUI/actions/tree/master/visual-diff).
```
Comment (margaree) [inline README.md:36]:
```suggestion
Visual diff results are published to a bucket in S3 and need special tokens to do so. To set these up, follow the instructions in the [visual-diff GitHub Action](https://github.com/BrightspaceUI/actions/tree/master/visual-diff).
```

**Review by margaree — APPROVED** [review]

File: src/generators/test-visual-diff/templates/visual-diff.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+      - uses: Brightspace/third-party-actions@actions/setup-node
+      - name: Install Dependencies
+        run: npm install
+      - name: Visual Diff Tests
+        uses: BrightspaceUI/actions/visual-diff@master
+        with:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_SESSION_TOKEN: ${{ secrets.AWS_SESSION_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (dlockhart) [inline src/generators/test-visual-diff/templates/visual-diff.yml:19]: I updated this to be its own workflow based on the experiment we ran in the navigation repo -- separate workflow allows the tests to run quickly in parallel, and if there are any failures we can re-run it easily.

**Review by svanherk — APPROVED** [review]
Looks good to me!

**Pull request merged**